src/websocket_stream.cpp

```diff
@@ -277,6 +286,35 @@ void websocket_stream::on_close(error_code)
 	COMPLETE_ASYNC("websocket_stream::on_close");
 }
 
+void websocket_stream::on_keepalive(error_code ec)
+{
+	if (ec) return;
+
```

Comment:
I think there's a risk for this to be called with `m_open` = `false`. It seems like that should be an early exit here.